### PR TITLE
Add iceberg to product tests suite

### PR DESCRIPTION
**Pull request by vrozov**
Fixes https://github.com/prestosql/presto/issues/2304

### CODE REVIEW

File: presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergCreateTable.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+import java.sql.SQLException;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tests.TestGroups.ICEBERG;
+import static io.prestosql.tests.TestGroups.STORAGE_FORMATS;
+import static io.prestosql.tests.utils.QueryExecutors.onPresto;
+
+public class TestIcebergCreateTable
```
Comment (kokosing) [inline presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergCreateTable.java]: It would be nice to have some generic connector tests, that we could run for hive and iceberg (and potentially for other connectors too in future).

Comment (vrozov) [reply presto-product-tests/src/main/java/io/prestosql/tests/iceberg/TestIcebergCreateTable.java]: The scope of the PR is to add iceberg connector to the product tests, so I limited scope of iceberg tests to a bare minimum (create/delete table and run simple query). I propose deferring generalization between hive and iceberg tests and extending iceberg tests scope to be on par with hive tests to a follow up PRs.

**Comment by vrozov** [comment]
> Is it on-pair with functionality with Hive connector

No, it is not and it was not the goal of the PR. The primary focus is to be mostly on par with various HDFS/HMS configurations that hive connector supports also for iceberg connector.

> There are plenty of tests failures.

Some tests are expected to fail as iceberg connector does not currently support Kerberized HDFS. I'll exclude iceberg tests from PR checks till support for Kerberized HDFS is added to iceberg connector.

**Comment by kokosing** [comment]
> Some tests are expected to fail as iceberg connector does not currently support Kerberized HDFS. I'll exclude iceberg tests from PR checks till support for Kerberized HDFS is added to iceberg connector.

If iceberg does not support kerberos then it should not be configured in kerberized environments.

**Comment by electrum** [comment]
It should supper Kerberos. We use the same HDFS code and bind everything the same way (at least that was the intention).

**Comment by vrozov** [comment]
Presto iceberg connector does not support Kerberos and the goal of the PR is to provide environments that may be used to fix the iceberg connector. There are a few improvements on the iceberg side that I implemented that will help iceberg connector to work properly with Kerberized HDFS.

**Comment by vrozov** [comment]
all product tests are fixed, please review.

File: presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/multinode-tls-kerberos/iceberg.properties [hunk 2]
```diff
@@ -0,0 +1,15 @@
+connector.name=iceberg
```
Comment (kokosing) [inline presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/multinode-tls-kerberos/iceberg.properties]: I would prefer to not to configure iceberg connector in environments where it does not work. That would be very misleading to other developers, where they would think where something just got broken because it does not work but it is committed.

**Comment by vrozov** [comment]
> Please also squash commits (otherwise first commit is obviously broken it has no chance to pass the automation).

Two commits are intentional as the first commit configures iceberg plugin for product tests suite. The second commit disables iceberg plugin in those environments where it will cause automation to fail.

File: presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/multinode-tls-kerberos/iceberg.properties (hunk 2 above)
Comment (vrozov) [reply presto-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/multinode-tls-kerberos/iceberg.properties]: IMO, there is a difference between committing broken functionality and providing a test environment that demonstrates that the functionality of a plugin is broken in certain configurations.

**Comment by vrozov** [comment]
@electrum Please review

**Comment by vrozov** [comment]
@electrum Please take a look

**Comment by electrum** [comment]
Thanks!
